go/enclave/events/subscription_manager.go

```diff
@@ -169,26 +169,26 @@ func (s *SubscriptionManager) encryptLogs(logsByID map[gethrpc.ID][]*types.Log)
 	return encryptedLogsByID, nil
 }
 
-// Extracts the (potential) user addresses from the topics. If there is no code associated with an address, it's a user
-// address.
+// Extracts the user addresses from the topics.
 func getUserAddrsFromLogTopics(log *types.Log, db *state.StateDB) []string {
-	var userAddrs []string //nolint:prealloc
+	var userAddrs []string
 
 	for _, topic := range log.Topics {
-		// Since addresses are 20 bytes long, while hashes are 32, only topics with 12 leading zero bytes can
-		// (potentially) be user addresses.
-		topicHex := topic.Hex()
-		if topicHex[2:len(zeroBytesHex)+2] != zeroBytesHex {
-			continue
-		}
+		potentialAddr := gethcommon.HexToAddress(topic.Hex())
 
-		// If there is code associated with an address, it is not a user address.
-		addr := gethcommon.HexToAddress(topicHex)
-		if db.GetCode(addr) != nil {
+		// A user address must have (at least) 12 leading zero bytes, since addresses are 20 bytes long, while hashes
+		// are 32.
+		if topic.Hex()[2:len(zeroBytesHex)+2] != zeroBytesHex {
 			continue
 		}
 
-		userAddrs = append(userAddrs, addr.Hex())
+		// A user address must exist in the state DB, and have a non-zero balance and nonce.
+		if db.Exist(potentialAddr) && db.GetBalance(potentialAddr).Cmp(big.NewInt(0)) != 0 && db.GetNonce(potentialAddr) != 0 {
```

Comment:
I think these shuold be || instead of &&

Reply:
I thought for a user address, all three had to be met? It has to exist with a real balance and nonce.

Reply:
it is possible to have no native balance remaining, but to have submitted transactions. 
or to have balance sent by a different account, in preparation to submit transactions

Reply:
I know. But I thought we were insisting on all three? If you remember, the 0x0000002 account had a balance, but no nonce.

Reply:
which means it's a genuine account. someone has sent money to that address

Reply:
we are using 2 somewhere for sure. I'll find it